lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -355,6 +355,47 @@ void printDecimalNumber(AsmPrinter &printer, DecimalType type,
   printer << "P = " << type.getPrecision() << ", S = " << type.getScale();
 }
 
+ParseResult parseAddRemovePadding(AsmParser &parser, StringAttr &value,
+                                  FixedBinaryType &type) {
+  // Parse fixed binary value as quoted string.
+  std::string valueStr;
+  if (parser.parseString(&valueStr))
+    return failure();
+
+  // Parse `L = <length>`.
+  uint32_t length;
+  if (parser.parseComma() || parser.parseInteger(length))
+    return failure();
+
+  // Create `FixedBinaryType`.
+  auto emitError = [&]() {
+    return parser.emitError(parser.getCurrentLocation());
+  };
+  if (!(type = FixedBinaryType::getChecked(emitError, parser.getContext(),
+                                           length)))
+    return failure();
+
+  // Pad or trim the value to match the fixed length.
```

Comment:
I am not sure if I like the padding. First, we can also create these attributes programmatically, for example during import, and those will not be padded. The padding of those wouldn't be removed during printing, but parsing that back in would then remove it, so they'd round-trip to a *different* attribute. In order to make that consistent, we'd have to verify that there is no padding and create convenience builders that remove the padding from arbitrary strings. Second, I only see a small benefit (a slightly more concise assembly for values that happen to end with `\0`) that also has the danger of bringing confusion (a mismatch between the apparent value length and the length specified in the type).